File: TransCoder_tokenized_test_set_functions/python/KTH_NON_REPEATING_CHARACTER.py
def kthNonRepeating ( str, k ) :
    n = len ( str )
    count = [ 0 ] * MAX_CHAR
    index = [ 0 ] * MAX_CHAR
    for i in range ( MAX_CHAR ) :
        count [ i ] = 0
        index [ i ] = n
    for i in range ( n ) :
        x = str [ i ]
        count [ ord ( x ) ] += 1
        if ( count [ ord ( x ) ] == 1 ) :
            index [ ord ( x ) ] = i
        if ( count [ ord ( x ) ] == 2 ) :
            index [ ord ( x ) ] = n
    index . sort ( )
    return index [ k - 1 ] if ( index [ k - 1 ] != n ) else - 1

File: TransCoder_tokenized_test_set_functions/cpp/KTH_NON_REPEATING_CHARACTER.cpp
int kthNonRepeating ( string str, int k ) {
  int n = str . length ( );
  int count [ MAX_CHAR ];
  int index [ MAX_CHAR ];
  for ( int i = 0;
  i < MAX_CHAR;
  i ++ ) {
    count [ i ] = 0;
    index [ i ] = n;
  }
  for ( int i = 0;
  i < n;
  i ++ ) {
    char x = str [ i ];
    ++ count [ x ];
    if ( count [ x ] == 1 ) index [ x ] = i;
    if ( count [ x ] == 2 ) index [ x ] = n;
  }
  sort ( index, index + MAX_CHAR );
  return ( index [ k - 1 ] != n ) ? index [ k - 1 ] : - 1;
}